packages/runtime/container-runtime/src/test/containerRuntime.spec.ts

```diff
@@ -1465,5 +1513,172 @@ describe("Runtime", () => {
 				]);
 			});
 		});
+
+		describe("Summarization", () => {
+			let containerRuntime: ContainerRuntime;
+
+			async function yieldEventLoop(): Promise<void> {
+				const yieldP = new Promise<void>((resolve) => {
+					setTimeout(resolve);
+				});
+				clock.tick(1);
+				await yieldP;
+			}
+
+			beforeEach(async () => {
+				const settings = {};
+				settings["Fluid.ContainerRuntime.Test.ValidateSummaryBeforeUpload"] = true;
+				containerRuntime = await ContainerRuntime.loadRuntime({
+					context: getMockContext(settings) as IContainerContext,
+					registryEntries: [],
+					existing: false,
+				});
+			});
+
+			it("summary is submitted successfully", async () => {
+				const summarizeResult = await containerRuntime.submitSummary({
```

Comment:
I'm amazed that this function is unit testable!

Reply:
I wish that function was simpler though. Hopefully, it will be once all our planned work is done.